src/controllers/AuthController.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response } from 'express';
+import { AuthService } from '../services/AuthService';
+
+export class AuthController {
+  private authService: AuthService;
+
+  constructor() {
+    this.authService = AuthService.getInstance();
+  }
+
+  public async login(req: Request, res: Response) {
+    const { email, password } = req.body;
+    
+    try {
+      const result = await this.authService.authenticateUser(email, password);
+      res.json(result);
+    } catch (error: any) {
```

Comment:
in general catching all errors and returning a 401 is a bit crude. meaning there may be more server side errors (or code issues) that trigger an exception that should be handled more precisely so we do not accidentally mislead the user the authentication failed when in fact it's a temporary problem on our end

Reply:
concretely, for server side errors we should opt for 5xx type codes.